tests/_files/tree_network_sim_expected_results/calc_pressure_loss.py

```diff
@@ -0,0 +1,182 @@
+
+
+for index, node in enumerate(mass_flow_total):
+    # Calculation of the velocity v
+    v[str(index)] = 4 * mass_flow_total[str(node)] \
+                    / (rho * pi * (pipes['diameter_mm'].iloc[index] / 1000) ** 2)
+
+    # Calculation of Re number
+    re[str(index)] = pipes['diameter_mm'].iloc[index] / 1000 * v[str(index)] * rho / (mu/1000)
+    # Calculation of lambda with simple approach
+    lambda_simp[str(index)] = 0.07 * re[str(index)]**-0.13 \
+                              * (pipes['diameter_mm'].iloc[index] / 1000)**-0.14
+    # Calculation of lambda with advanced approach
+    lambda_adv[str(index)] = 1.325 / \
+                             np.log(epsilon/(1000*3.7*(pipes['diameter_mm'].iloc[index] / 1000))
+                                            + 5.74/(re[str(index)]**0.9))**2
+    # Calculate distributed pressure losses with Darcy-Weissbach-equation
+    dp_diss[str(index)] = lambda_simp[str(index)] * rho * \
+                          pipes['length_m'].iloc[index] * v[str(index)]**2 / \
+                          (2 * (pipes['diameter_mm'].iloc[index] / 1000))
+    # Calculate local pressure losses resulted from separating Tee (T-Stück) -> i - inlet
+    # Localized Pressure losses only occur in the outlet pipes of the tee separator
+    if node == '0':
+        dp_loc_tee_i[str(index)] = 0 * zeta_tee_separation * v[str(index)] ** 2 * rho / 2
+    elif node != '0':
+        dp_loc_tee_i[str(index)] = zeta_tee_separation * v[str(index)] ** 2 * rho / 2
+    # Calculate local pressure losses resulted from connecting Tee (T-Stück) -> r - return
+    # Localized Pressure losses only occur in the outlet pipes of the tee connector
+    if node == '0':
+        dp_loc_tee_r[str(index)] = zeta_tee_connect * v[str(index)] ** 2 * rho / 2
+    elif node != '0':
+        dp_loc_tee_r[str(index)] = 0 * zeta_tee_connect * v[str(index)] ** 2 * rho / 2
+    # Calculate local pressure losses resulted from consumer valves
+    if node == '0':
+        dp_loc_valve[str(index)] = 0 * v[str(index)] ** 2 * rho / 2   # Producer no valve, zeta = 0
+    elif node != '0':
+        dp_loc_valve[str(index)] = zeta_valve * v[str(index)] ** 2 * rho / 2
+
+# Calculate distributed pressure losses for inlet and return direction (dp_diss * 2)
+dp_diss = dp_diss * 2
+
+# Calculate sum of local pressure losses
+dp_loc_tee = dp_loc_tee_i + dp_loc_tee_r
+dp_loc = dp_loc_tee
+
+# Calculate hydrostatic pressure difference
+dp_hyd['0'] = - rho * g * abs(producers['m_over_NHN'][0] - forks['m_over_NHN'][0]) * v['0']**0
+dp_hyd['1'] = - rho * g * abs(forks['m_over_NHN'][0] - consumers['m_over_NHN'].iloc[0]) * \
+              v['1']**0
+dp_hyd['2'] = - rho * g * abs(forks['m_over_NHN'][0] - consumers['m_over_NHN'].iloc[1]) * \
+              v['2']**0
+
+# Calculate total pressure loss
+dp = dp_diss + dp_loc + dp_hyd
+
+# Find the consumer with maximum pressure losses
+max_pressure_losses = max(dp.loc[:, '1':])
```

Comment:
This seems to only select the pressure loss in the branch of the consumer. In this example it may make no difference to the results, but the whole path from producer to consumer with the highest pressure losses should be chosen.

Reply:
So it should be "max_pressure_losses = max(dp.loc[:, '0':])". Is that correct?